skaha/src/main/java/org/opencadc/skaha/session/SessionServiceBuilder.java

```diff
@@ -0,0 +1,95 @@
+package org.opencadc.skaha.session;
+
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1OwnerReference;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.openapi.models.V1ServiceSpec;
+import io.kubernetes.client.util.Yaml;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.opencadc.skaha.Job;
+import org.opencadc.skaha.SessionType;
+
+/**
+ * Programmatically build a Kubernetes service for a session, using the session type and job information to provide a
+ * base YAML file. Example usage:
+ *
+ * <pre>
+ *     final Job job = new Job("name", "uid", "my-sessionID", SessionType.CARTA);
+ *     final SessionServiceBuilder sessionServiceBuilder = new SessionServiceBuilder(job);
+ *
+ *     // Can be used to execute in a Kubernetes cluster.
+ *     final String outputYAML = sessionServiceBuilder.build();
+ * </pre>
+ */
+public class SessionServiceBuilder {
+    private static final String RUN_LABEL_KEY = "run";
+    private static final String SELECTOR_KEY = "canfar-net-sessionID";
+
+    private final Job job;
+
+    public SessionServiceBuilder(final Job job) {
+        this.job = Objects.requireNonNull(job);
```

Comment:
Raise a proper error message here, `this.job = Objects.requireNonNull(job, "Job cannot be null");`